From 306dcbe2a389e469124ac606aa7236ca83b55dbf Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Tue, 9 Jan 2024 16:58:40 +0100
Subject: [PATCH 0910/1141] ARM: dts: stm32: add wake-up button on stm32mp25
 boards

Adds wake-up button, that comes from OPTEE interrupt number ZERO.

Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I10b522b90640518c311b4a78c1933d51db53f48b
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/350297
Reviewed-by: Pascal PAILLET <p.paillet@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Pascal PAILLET <p.paillet@st.com>
---
 arch/arm64/boot/dts/st/stm32mp257f-dk.dts  | 7 +++++++
 arch/arm64/boot/dts/st/stm32mp257f-ev1.dts | 7 +++++++
 2 files changed, 14 insertions(+)

--- a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
@@ -86,6 +86,13 @@
 			gpios = <&gpioc 11 GPIO_ACTIVE_HIGH>;
 			status = "okay";
 		};
+
+		button-wake-up {
+			label = "wake-up";
+			linux,code = <KEY_WAKEUP>;
+			interrupts-extended = <&optee 0>;
+			status = "okay";
+		};
 	};
 
 	gpio-leds {
--- a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
@@ -64,6 +64,13 @@
 			gpios = <&gpiod 2 GPIO_ACTIVE_HIGH>;
 			status = "okay";
 		};
+
+		button-wake-up {
+			label = "wake-up";
+			linux,code = <KEY_WAKEUP>;
+			interrupts-extended = <&optee 0>;
+			status = "okay";
+		};
 	};
 
 	gpio-leds {
